checkstyle-configuration.xml

```diff
@@ -0,0 +1,175 @@
+<?xml version="1.0"?>
+<!DOCTYPE module PUBLIC
+    "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
+    "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
+<!--
+    Copyright (c) 2011-present Sonatype, Inc. All rights reserved.
+    Includes the third-party code listed at http://links.sonatype.com/products/clm/attributions.
+    "Sonatype" is a trademark of Sonatype, Inc.
+-->
+<module name="Checker">
+  <property name="charset" value="UTF-8"/>
+
+  <property name="severity" value="error"/>
+
+  <property name="fileExtensions" value="java"/>
+  <module name="FileTabCharacter">
+    <property name="eachLine" value="true"/>
+  </module>
+
+  <module name="NewlineAtEndOfFile"/>
+
+  <module name="SuppressWarningsFilter" />
+
+  <module name="TreeWalker">
+    <module name="SuppressWarningsHolder" />
+    <module name="OuterTypeFilename"/>
+    <module name="LineLength">
+      <property name="max" value="120"/>
+      <!-- Allow longer lines in comments. -->
```

Comment:
Why would we allow longer lines in comments?

Reply:
Well, https://github.com/sonatype/codestyle/blob/checkstyle/src/main/java/Comments.java has lines longer than 120 characters in the comments and in a location where it'd be hard to use a specific annotations based override.  So, it seemed better to allow long comments than have no line length enforcement at all.

We could implement http://checkstyle.sourceforge.net/config_filters.html in addition or instead of the @SuppressWarnings annotation to allow conscious rule breaking.